File: server/src/main/java/umm3601/Family/Family.java
// Packages
package umm3601.Family;

// Java Imports
import java.util.List;

// Org Imports
import org.mongojack.Id;
import org.mongojack.ObjectId;

@SuppressWarnings({"VisibilityModifier"})
public class Family {
  @ObjectId @Id
  @SuppressWarnings({"MemberName"})
  public String _id;

  public String guardianName;
  public String email;
  public String address;
  public String timeSlot;

  public List<StudentInfo> students;

  public static class StudentInfo {
    public String name;
    public String grade;
    public String school;
    public List<String> requestedSupplies;
  }
}

File: client/src/app/family/family.ts
export interface StudentInfo {
  name: string;
  grade: string;
  school: string;
  requestedSupplies: string[];
}

export interface Family {
  _id?: string;
  guardianName: string;
  email: string;
  address: string;
  timeSlot: string;
  students: StudentInfo[];
}

export interface DashboardStats {
  studentsPerSchool: { [school: string]: number};
  studentsPerGrade: { [grade: string]: number};
  totalFamilies: number;
}
